src/compiler/crystal/codegen/primitives.cr

```diff
@@ -1076,7 +1076,27 @@ class Crystal::CodeGenVisitor
     codegen_tuple_indexer(context.type, call_args[0], index)
   end
 
-  def codegen_tuple_indexer(type, value, index)
+  def codegen_tuple_indexer(type, value, index : Range)
+    case type
+    when TupleInstanceType
+      tuple_types = type.tuple_types[index].map &.as(Type)
```

Comment:
It's so convenient that implementing `[]` with range is actually done by using `[]` with a range :-)

When I initially thought about this I thought it would be more complex because of all the indexing logic involved... but it's already there!